### DISCUSSION TITLE
pg_yregress: possible to have setup script?

### DISCUSSION THREAD

**Discussion by ccleve** [omnigres/omnigres#289] [General]
My tests all depend on a setup script that installs an extension, creates a table, and populates it. Is it possible to get pg_yregress to execute that script before each test or group of tests?

Most test frameworks have some kind of setup() and teardown() functionality.

**Comment by yrashk** [comment]
For a general case, I would suggest grouping setups using [`instance->init`](https://docs.omnigres.org/pg_yregress/usage/#single-instance-configuration) (a subset of [`instances`](https://docs.omnigres.org/pg_yregress/usage/#configuring-instances))

There is no teardown functionality (yet)

Every test is wrapped in a rolled-back transaction by default.

Would you mind sharing what exactly your teardown script does?

**Comment by ccleve** [comment]
There's no real need for a teardown script. It would be helpful to drop some indexes and extensions, but you can do that at the beginning of your setup script with IF EXISTS. How time-consuming would it be to init a new instance before each test group?

**Reply by yrashk** [reply]
Generally, it doesn't take a lot of time to create and start an instance, sub-second type of deal, assuming no slow .so preload specified in the config. Instances are only created per file (one by default or more if `instances` is configured); individual tests in them are simply transactions within that.
